src/pineko/comparator.py

```diff
@@ -32,19 +34,52 @@ def compare(pine, fktable, max_as, max_al, pdf, xir, xif):
     """
     import lhapdf  # pylint: disable=import-error,import-outside-toplevel
 
-    pdfset = lhapdf.mkPDF(pdf, 0)
-    pdgid = int(pdfset.set().get_entry("Particle"))
+    pdfset1 = lhapdf.mkPDF(pdf1, 0)
+    pdgid1 = int(pdfset1.set().get_entry("Particle"))
+
+    if pdf2 is not None:
+        pdfset2 = lhapdf.mkPDF(pdf1, 0)
+        pdgid2 = int(pdfset2.set().get_entry("Particle"))
+    else:
+        pdfset2 = pdfset1
+        pdgid2 = pdgid1
+
+    try:
+        parton1 = pine.key_values()["convolution_particle_1"]
+        parton2 = pine.key_values()["convolution_particle_2"]
+    except KeyError:
+        parton1 = pine.key_values()["initial_state_1"]
+        parton2 = pine.key_values()["initial_state_2"]
+    hadronic = parton1 == parton2
+
     order_mask = pineappl.grid.Order.create_mask(pine.orders(), max_as, max_al, True)
-    before = np.array(
-        pine.convolute_with_one(
-            pdgid,
-            pdfset.xfxQ2,
-            pdfset.alphasQ2,
-            order_mask=order_mask,
-            xi=((xir, xif),),
+    if hadronic:
+        before = np.array(
+            pine.convolve_with_two(
+                pdg_id1=pdgid1,
```

Comment:
done in 2553fe1

Reply:
My point still holds: you still need `pdgid1 = int(pdfset1.set().get_entry("Particle"))`. To clarify: my worries apply here and also to the evolution side in a similar way. Maybe it is easiest if I explain by example:
1. consider a grid with an unpolarized PDF under "1" and a polarized PDF under "2"
2. then, in order to use this function (and `evolve`), you have to a) know that beforehand and b) pass exactly a unpol. PDF as 1 and a pol. PDF as 2 and never the other way up

this might be fine for the moment, because we are the only ones generating grids and we, meaning you and @Radonirinaunimi , made that silent agreement - but if we want to be open then we either need to expose that information, or check for that convention.

Actually, now that I thought 2 more minutes on the problem: maybe for LHAPDF there is nothing we can do, because we can not know from the object whether it is polarized or not, can we? But for the evolution side we can know and we should check. If you wish, I can move the comment down there if you agree that for LHAPDF we can do nothing but hope for the best.

Reply:
okay I see your point. So in LHAPDF there is no way to discriminate the polarisation if not in the info comments or from the name... 
We can add a check that the evolution and the grid metadata are in agreement, to on the safer side.

Reply:
Absolutely agree with @felixhekhorn's point. We should not always assume that the convolution types of the grids are ordered in the same way, nor the ordering of the PDFs to be convoluted with. At least from the `n3fit` and `vp` side, I try to account for this as much as possible.

Reply:
what about 04a56f7 ?